node_modules/@stdlib/iter/cartesian-power/README.md

```diff
@@ -0,0 +1,212 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2024 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# iterCartesianPower
+
+> Create an iterator which generates the Cartesian power of an input array-like object.
+
+<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->
+
+<section class="intro">
+
+</section>
+
+<!-- /.intro -->
+
+<!-- Package usage documentation. -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var iterCartesianPower = require( '@stdlib/iter/cartesian-power' );
+```
+
+#### iterCartesianPower( x, n )
+
+Returns an iterator which generates the Cartesian power of an input array-like object.
+
+```javascript
+var x = ['a', 'b', 'c'];
+var n = 2;
+var iterator = iterCartesianPower( x, n );
+
+for (const combination of iterator) {
+    console.log( combination );
+}
+
+```
+
+</section>
+
+<!-- /.usage -->
+
+<!-- Package usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="notes">
+
+## Notes
+
+- The function expects the first argument x to be an array-like object.
+- The second argument n must be a non-negative integer.
+- The returned iterator will generate all possible combinations of n elements from the input array x.
+    
+</section>
+
+<!-- /.notes -->
+
+<!-- Package usage examples. -->
+
+<section class="examples">
+
+## Examples
+
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+// Example 1: Generating Cartesian power of an array with n = 2
+var x = [1, 2, 3];
+var n = 2;
+var iterator = iterCartesianPower(x, n);
+
+for (const combination of iterator) {
+    console.log(combination);
+}
+
+// Output:
+// [ 1, 1 ]
+// [ 1, 2 ]
+// [ 1, 3 ]
+// [ 2, 1 ]
+// [ 2, 2 ]
+// [ 2, 3 ]
+// [ 3, 1 ]
+// [ 3, 2 ]
+// [ 3, 3 ]
+
+```
+
+```javascript
```

Comment:
Don't split up examples like this. This does not match the conventions in this project.